### PR TITLE
Added support for Labels to be set from AppSettings.json or web/app.config

### PR DESCRIPTION
**Pull request by quigamdev**
## Description
I have added support for Labels to be added/modified via AppSettings.json or app/web.config file. 
Our internal system does not allow to change of process variables or modification of other files than appsettings.json (cannot use newrelic.config).
So I cannot use the currently supported ways.

# Author Checklist
- [x] Unit tests, Integration tests, and Unbounded tests completed
- [x] Performance testing completed with satisfactory results (if required) - should not be required for this PR
- [x] [Agent Changelog](/src/Agent/CHANGELOG.md) or [Lambda Agent changelog](/src/AwsLambda/CHANGELOG.md) updated 

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)
- [ ] Review Changelog

### CODE REVIEW

**Comment by JcolemanNR** [comment]
Hey @quigamdev, thanks for the PR to add this capability! The team discussed your PR today, and we would like to move forward with getting it merged/released with some minor changes (mostly unit tests). I'll add some comments/references to comparable tests.

Don't worry about any CI failures you see from this PR, there is an issue we're working to resolve an issue with some tooling that breaks when PRs are submitted from forks.

Before this is merged we'll need you to sign the CLA. See the helpful comment from the CLAassistant above. 👍

File: src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs
Comment (JcolemanNR) [inline src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs:1300]:
```suggestion
                    Log.Info("Application labels from web.config, app.config, or appsettings.json.");
```

File: src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs
Comment (JcolemanNR) [inline src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs:1297]: A pattern we're moving towards (in the interest of performance) is caching values like this in private class members after they're calculated the first time to prevent repeatedly querying ConfigurationManager and Environment Variables. 

In this case, could you introduce a private member named `_labels`, to avoid un-necessarily querying ConfigurationManager after the first time this Property getter is called?

File: src/Agent/CHANGELOG.md
Comment (JcolemanNR) [inline src/Agent/CHANGELOG.md:10]:
```suggestion
* Support of setting up labels via appsettings.json and app/web.config file. Community contribution by [quigamdev](https://github.com/quigamdev)! [#1204](https://github.com/newrelic/newrelic-dotnet-agent/pull/1204)
```

File: src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs
Comment (JcolemanNR) [inline src/Agent/NewRelic/Agent/Core/Configuration/DefaultConfiguration.cs:1295]: This accessor is unit tested [HERE](https://github.com/newrelic/newrelic-dotnet-agent/blob/e925389eaff3790381467749c71250c586baa194/tests/Agent/UnitTests/Core.UnitTest/Configuration/DefaultConfigurationTests.cs#L1299-L1309), and should be extended to test the the new app local config override functionality you are adding.

I would suggest matrixing in a parameter to test that the ConfigurationManager based override works as expected. Examples on how to mock ConfigurationManger can be found in the same file.

**Review by JcolemanNR — CHANGES_REQUESTED** [review]
Thanks again for the submission. Let's get the CLA signed, and the requested changes implemented, and we can merge this! Let me know if you want any additional guidance or help implementing the requested changes. 🎉 🚀

**Comment by quigamdev** [comment]
I'm sorry that I wasn't able to get to it earlier. I have been busy with other projects. Thank you for finishing it. 🙏🏻

**Pull request merged**
